src/Plugin/EntityMetaRelation/ListPage.php

```diff
@@ -160,13 +176,52 @@ public function build(array $form, FormStateInterface $form_state, ContentEntity
       ],
     ];
 
-    if (!empty($form_state->getValue('entity_type')) || $entity_bundle_id) {
+    $entity_bundle_id = $entity_meta_wrapper->getSourceEntityBundle();
+
+    if (!empty($selected_entity_type) || $entity_bundle_id) {
       $form[$key]['bundle_wrapper']['bundle'] = [
         '#type' => 'select',
         '#title' => $this->t('Source bundle'),
-        '#default_value' => $form_state->getValue('bundle') ?? $entity_bundle_id,
+        '#default_value' => $form_state->getValue('bundle', $entity_bundle_id),
         '#options' => $bundle_options,
         '#required' => TRUE,
+        '#ajax' => [
+          'callback' => [$this, 'updateExposedFilters'],
+          'disable-refocus' => FALSE,
+          'event' => 'change',
+          'wrapper' => 'list-page-exposed-filters',
+        ],
+      ];
+    }
+
+    $form[$key]['bundle_wrapper']['exposed_filters_wrapper'] = [
+      '#tree' => TRUE,
+      '#type' => 'container',
+      '#attributes' => [
+        'id' => 'list-page-exposed-filters',
+      ],
+    ];
+
+    $selected_entity_type = $form[$key]['entity_type']['#default_value'] ?? NULL;
+    $selected_bundle = $form[$key]['bundle_wrapper']['bundle']['#default_value'] ?? NULL;
```

Comment:
There is a problem here: if you change the entity type $form[$key]['bundle_wrapper']['bundle']['#default_value'] will still have the value of the previously selected entity type, which in turn means  that we never get the appropriate list source to generate the filters.

Reply:
You can reproduce this by first configuring a site to have facets for a node content type. Once you have that, when you go to the node creation form you will see the list page meta form fields: entity, bundle and in this case, the filters.
If you select a different entity type the form will get updated and show the bundles of that entity type and no filters (this is what we want since that entity type won't have any facets). However when you go back to the node content type, when the form gets updated you won't see any filters.

Reply:
This problem is still not resolved.